### PR TITLE
Ignite source-map

### PR DESCRIPTION
**Pull request by GantMan**
Add source-map reader to Ignite

image

image

### CODE REVIEW

File: packages/ignite-cli/src/lib/exitCodes.js [hunk 1]
```diff
@@ -65,5 +65,15 @@ module.exports = {
   /**
    * One of the global dependencies are not met.
    */
-  INVALID_GLOBAL_DEPENDENCY: 12
+  INVALID_GLOBAL_DEPENDENCY: 12,
+
+  /**
+   * There is no 13th floor.. only Zuul
+   */
+  TRISKAIDEKAPHOBIA_DETECTED: 13,
```
Comment (skellock) [inline packages/ignite-cli/src/lib/exitCodes.js]: LOL!!!!

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
   "repository": "infinitered/ignite",
   "license": "MIT",
   "devDependencies": {
-    "lerna": "2.0.0-beta.38"
+    "lerna": "2.0.0-beta.38",
+    "source-map": "^0.5.6"
```
Comment (skellock) [inline package.json]: You won't want this here.  You'll want this in the package.

File: packages/ignite-cli/src/commands/sourceMap.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const SourceMapConsumer = require('source-map').SourceMapConsumer
+const exitCodes = require('../lib/exitCodes')
+
+const printSourceResult = (smc, position, colors) => {
+  const {source, line, column, name} = smc.originalPositionFor(position)
+  const styledName = colors.red(name)
+  const styledSource = colors.yellow(source)
+
+  console.log(`${styledName}: ${styledSource}
```
Comment (skellock) [inline packages/ignite-cli/src/commands/sourceMap.js]: I'd recommend using `context.print.info`.  This means you'll have to put this function in the main `module.exports` function.

File: packages/ignite-cli/src/commands/sourceMap.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  const line = options.line || options.l
+  const column = options.column || options.c
+  const stackfile = options.stackfile || options.s
+
+  // make sure mapfile is passed with line+column || stackfile supplied
+  const optionsValid = (mapfile && ((line && column) || stackfile))
+  if (!optionsValid) {
+    error(`You must pass a mapfile AND either a line+column or a stackfile.
+    e.g.
+    ignite source-map --map ./someFolder/index.ios.map --line 32 --column 76
```
Comment (skellock) [inline packages/ignite-cli/src/commands/sourceMap.js]: How would you feel about upgrading the required fields to parameters from options?

```sh
# a line
ignite source-map ./someFolder/index.ios.map 32

# a line + column
ignite source-map ./someFolder/index.ios.map 32:76

# a stackfile
ignite source-map ./someFolder/index.ios.map stackTrace.txt

# an error
ignite source-map ./someFolder/index.ios.map
```

**Comment by GantMan** [comment]
Closed in favor of:  https://github.com/GantMan/ignite-source-map

File: packages/ignite-cli/src/lib/exitCodes.js (hunk 1 above)
Comment (jamonholmgren) [inline packages/ignite-cli/src/lib/exitCodes.js]: I'm really really disappointed this isn't in our current code base @GantMan ... sneak it in your next PR!

**Pull request closed without merging**